### PR TITLE
redhat: adjust IP bonding to bond0 only

### PR DESCRIPTION
**Pull request by dustinmiller**
It's possible that adding IPs to bond1 can create a scenario where 
networking isn't happy. Lets remove those accordingly

### CODE REVIEW

File: packetnetworking/distros/redhat/templates/bonded/etc_sysconfig_network-scripts_ifcfg-bondX.j2 [hunk 1]
```diff
@@ -1,5 +1,6 @@
 DEVICE={bond}
 NAME={bond}
+{{% if bond == "bond0" %}}
```
Comment (nuclearbob) [inline packetnetworking/distros/redhat/templates/bonded/etc_sysconfig_network-scripts_ifcfg-bondX.j2:3]:
```suggestion
{{% if "{bond}" == "bond0" %}}
```

File: packetnetworking/distros/redhat/templates/bonded/etc_sysconfig_network-scripts_ifcfg-bondX.j2 [hunk 2]
```diff
@@ -9,17 +10,20 @@ IPADDR={{{{ ip4priv.address }}}}
...
 GATEWAY={{{{ ip4priv.gateway }}}}
 {{% endif %}}
+{{% endif %}}
 BOOTPROTO=none
 ONBOOT=yes
 USERCTL=no
 TYPE=Bond
 BONDING_OPTS="mode={{{{ net.bonding.mode }}}} miimon=100 downdelay=200 updelay=200"
 
+{{% if bond == "bond0" %}}
```
Comment (nuclearbob) [inline packetnetworking/distros/redhat/templates/bonded/etc_sysconfig_network-scripts_ifcfg-bondX.j2:20]:
```suggestion
{{% if "{bond}" == "bond0" %}}
```

**Review by nuclearbob — CHANGES_REQUESTED** [review]
bond isn't a variable in the jinja environment, it gets replaced in a formatting pass before the template is rendered. These changes allowed the tests to pass locally for me.

**Review by nuclearbob — APPROVED** [review]
Ship it!

File: packetnetworking/distros/redhat/templates/bonded/etc_sysconfig_network-scripts_ifcfg-bondX.j2 [hunk 3]
```diff
@@ -1,5 +1,6 @@
 DEVICE={bond}
 NAME={bond}
+{{% if "{bond}" == "bond0" %}}
```
Comment (mikemrm) [inline packetnetworking/distros/redhat/templates/bonded/etc_sysconfig_network-scripts_ifcfg-bondX.j2:3]:
```suggestion
{{% if bond == "bond0" %}}
```

**Review by mikemrm — CHANGES_REQUESTED** [review]
Other than this, and testing changes. looks good

File: packetnetworking/distros/redhat/templates/bonded/etc_sysconfig_network-scripts_ifcfg-bondX.j2 (hunk 3 above)
Comment (mikemrm) [reply packetnetworking/distros/redhat/templates/bonded/etc_sysconfig_network-scripts_ifcfg-bondX.j2:3]: 🤦 this is python formatted before it's handled with jinja2, so this is proper

**Review by mikemrm — APPROVED** [review]
once live tests have been completed, 👍

**Comment by nuclearbob** [comment]
Testing on actual machines went well except for n3.xlarge.x86, which shows problems on alma_8, rocky_8, and centos_8. alma and rocky install correctly with current osie, so I'm looking into what goes wrong when we use the new one.

**Comment by nuclearbob** [comment]
alma_8 is now broken with or without this, so let's go ahead and land it, and I'll keep working on alma.

**Comment by nuclearbob** [comment]
Oh, let's get a review from another team? Maybe @nshalman can recommend someone?

**Comment by nuclearbob** [comment]
Do we still want a review from another team on this, or is it okay to go in since it's a fix requested by another team and it's pretty small? Either way, I'd like to keep moving on this since it fixes issues in some of the plans we're enabling.

**Pull request merged**